### PR TITLE
feat(js): Support getResponseHeader in useApiQuery

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This allows us to extract headers from the API result when using useApiQuery

Important since we have a number of places where we pull out page links

### CODE REVIEW

File: static/app/views/settings/projectSourceMaps/projectSourceMapsArtifacts.tsx [hunk 1]
```diff
@@ -127,48 +127,41 @@ export function ProjectSourceMapsArtifacts({params, location, router, project}:
...
-    );
+        query: {query, cursor},
+      },
+    ],
+    {
+      staleTime: 0,
+      keepPreviousData: true,
+      enabled: tabDebugIdBundlesActive,
+    }
+  );
```
Comment (evanpurkhiser) [inline static/app/views/settings/projectSourceMaps/projectSourceMapsArtifacts.tsx:164]: Much cleaner

**Review by malwilley — COMMENTED** [review]
Nice! Might want to add a test case?

**Comment by evanpurkhiser** [comment]
Good call!

**Comment by malwilley** [comment]
Seems to have broken some existing usages like saved searches?

**Review by malwilley — APPROVED** [review]

**Review by scttcper — APPROVED** [review]

**Pull request merged**